plans/upgrade.pp

```diff
@@ -1,18 +1,16 @@
 plan autope::upgrade(
-  TargetSpec                           $targets          = get_targets('peadm_nodes'),
-  String                               $version          = '2019.3.0',
-  String                               $ssh_user,
-  Enum['xlarge', 'large', 'standard']  $architecture     = 'xlarge',
-  Enum['google', 'aws', 'azure']       $provider         = 'google'
+  String                               $version             = '2021.0.0',
+  Integer                              $compiler_count      = 1,
+  Enum['google', 'aws', 'azure']       $provider            = 'aws',
+  String[1]                            $ssh_user            = $provider ? { 'aws' => 'centos', default => undef },
 ) {
 
   Target.new('name' => 'localhost', 'config' => { 'transport' => 'local'})
 
   $tf_dir = ".terraform/${provider}_pe_arch"
 
-  if $provider == 'aws' {
-    waring('AWS provider is currently expiremental and may change in a future release')
-  }
+  $terraform_output = run_task('terraform::output', 'localhost', dir => $tf_dir)
+  $applied = $terraform_output.first
```

Comment:
$applied is a bit confusing since I believe it just contains the first result from the result set coming from the terraform::output task?

we could also write 
```
$terraform_output = run_task('terraform::output', 'localhost', dir => $tf_dir).first
``` 
and use $terraform_output further in the plan.

Reply:
Good idea! 👍